contracts/ColdStaking.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.4.18;
+
+import './safeMath.sol';
+
+contract ColdStaking {
+
+    using SafeMath for uint256;
+
+    event StartStaking(address addr, uint256 value, uint256 weight, uint256 init_block);
+
+    event WithdrawStake(address staker, uint256 weight);
+
+    event Claim(address staker, uint256 reward);
+
+    event FirstStakeDonation(address _address, uint256 value);
+
+
+    struct Staker
+    {
+        uint256 weight;
+        uint256 init_block;
+    }
+
+    uint256 public staking_pool;
+
+    uint256 public staking_threshold = 0 ether;
+
+    //uint256 public round_interval    = 172800; // approx. 1 month in blocks
+    //uint256 public max_delay      = 172800 * 12; // approx. 1 year in blocks
+
+
+    /// TESTING VALUES
+    uint256 public round_interval = 200; // approx. 1 month in blocks
+    uint256 public max_delay = 7 * 6000; // approx. 1 year in blocks
+
+    mapping(address => Staker) staker;
+    mapping(address => bool) private muted;
+
+
+    function() public payable
+    {
+        // No donations accepted to fallback!
+        // Consider value deposit is an attempt to become staker.
+        start_staking();
+    }
+
+    function start_staking() public payable
+    {
+        assert(msg.value >= staking_threshold);
+        staking_pool = staking_pool.add(msg.value);
+
+        Staker storage _staker = staker[msg.sender];
+
+        _staker.weight = _staker.weight.add(msg.value);
```

Comment:
start_staking wasn't increasing staker weight nor adding value to staking_pool

Reply:
It seems that it is related to the updated logic of `safeMath.sol` contract.